qiskit/circuit/quantumcircuit.py

```diff
@@ -2267,18 +2291,52 @@ def __getitem__(self, item):
         return self._data[item]
 
     @staticmethod
+    @deprecate_func(
+        since=1.2,
+        removal_timeline="in qiskit 2.0",
+        additional_msg="This method is only used as an internal helper "
+        "and will be removed with no replacement.",
+    )
     def cast(value: S, type_: Callable[..., T]) -> Union[S, T]:
         """Best effort to cast value to type. Otherwise, returns the value."""
         try:
             return type_(value)
         except (ValueError, TypeError):
             return value
 
+    @staticmethod
+    def _cast(value: S, type_: Callable[..., T]) -> Union[S, T]:
+        """Best effort to cast value to type. Otherwise, returns the value."""
+        try:
+            return type_(value)
+        except (ValueError, TypeError):
+            return value
+
+    @deprecate_func(
+        since=1.2,
+        removal_timeline="in qiskit 2.0",
+        additional_msg="This method is only used as an internal helper "
+        "and will be removed with no replacement.",
+    )
     def qbit_argument_conversion(self, qubit_representation: QubitSpecifier) -> list[Qubit]:
         """
         Converts several qubit representations (such as indexes, range, etc.)
         into a list of qubits.
 
+        Args:
+            qubit_representation (Object): representation to expand
+
+        Returns:
+            List(Qubit): the resolved instances of the qubits.
```

Comment:
Since the typehints are inline in the signature, I don't think they are needed in the docstring anymore 🙂 (Same below for the `clbit_argument_conversion`)
```suggestion
            qubit_representation: Representation to expand.

        Returns:
            The resolved instances of the qubits.
```